photon-core/src/test/java/org/photonvision/common/configuration/ConfigTest.java

```diff
@@ -141,4 +142,18 @@ public void testJacksonHandlesOldVersions() throws IOException {
 
         new File("test.json").delete();
     }
+
+    @Test
+    public void testJacksonHandlesOldTargetEnum() throws IOException {
+        var str =
+                "[ \"AprilTagPipelineSettings\", {\n" + "  \"targetModel\" : \"k6in_16h5\"\n" + "} ]\n";
+        var writer = new FileWriter("test.json");
+        writer.write(str);
+        writer.flush();
+        writer.close();
+        Assertions.assertDoesNotThrow(
+                () -> JacksonUtils.deserialize(Path.of("test.json"), CVPipelineSettings.class));
```

Comment:
What I would do here is cast the result to an ApriltagPipelienSettings so you can check it got loaded right

Reply:
Can I add another assertion inside the lambda?

Reply:
Since throwing an exception will cause the test to fail. I removed the `assertDoesNotThrow` and added an `assertEquals` on the deserialized value of targetModel.